src/order/succ_pred/relation.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2022 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import order.succ_pred.basic
+/-!
+# Relations on types with a `succ_order`
+
+This file contains properties about relations on types with a `succ_order`
+and their closure operations (like the transitive closure).
+-/
+
+open set relation succ_order function
+
+variables {α : Type*} [linear_order α] [succ_order α] [is_succ_archimedean α] [no_max_order α]
+/-- `(n, m)` is in the reflexive-transitive closure of `~` if `i ~ i+1` and `i+1 ~ i`
+  for all `i` between `n` and `m`. -/
+lemma refl_trans_gen_of_succ (r : α → α → Prop) {n m : α}
+  (h1 : ∀ i ∈ Ico n m, r i (succ i)) (h2 : ∀ i ∈ Ico m n, r (succ i) i) : refl_trans_gen r n m :=
```

Comment:
What I meant is that there are two natural statements to get out of this:
```suggestion

/-- If `i` is related to `succ i` for all `i` between `a` and `b`, then the transitive closure of
`~` relates `a` and `b` if `a < b`. -/
lemma trans_gen_of_succ_of_lt (r : α → α → Prop) {a b : α}  (h : ∀ i ∈ Ico a b, r i (succ i))
  (hab : a < b) : trans_gen r a b :=
sorry

/-- If `succ i` is related to `i` for all `i` between `a` and `b`, then the transitive closure of
`~` relates `b` and `a` if `a < b`. -/
lemma trans_gen_of_succ_of_gt (r : α → α → Prop) {a b : α}  (h : ∀ i ∈ Ico a b, r (succ i) i)
  (hab : a < b) : trans_gen r b a :=
sorry -- Use `trans_gen_of_succ_of_lt` and `swap (trans_gen r) = trans_gen (swap r)`

/-- If `i` and `i + 1` are related for all `i` between `a` and `b`, then the transitive closure
of `~` relates `a` and `b` if `a ≠ b`. -/
lemma trans_gen_of_succ_of_ne (r : α → α → Prop) {a b : α} (h1 : ∀ i ∈ Ico a b, r i (succ i))
  (h2 : ∀ i ∈ Ico b a, r (succ i) i) (hnm : a ≠ b) :
  trans_gen r a b :=
sorry -- use `trans_gen_of_succ_of_lt` and `trans_gen_of_succ_of_gt`

/-- If `i` and `i + 1` are related for all `i` between `a` and `b`, then the reflexive transitive
closure of `~` relates `a` and `b`. -/
lemma refl_trans_gen_of_succ (r : α → α → Prop) {n m : α}
  (h1 : ∀ i ∈ Ico n m, r i (succ i)) (h2 : ∀ i ∈ Ico m n, r (succ i) i) : refl_trans_gen r n m :=
sorry -- use `trans_gen_of_succ_of_ne`
```

Reply:
I see what you mean. Using `swap` is nice!